### PR TITLE
add bucket logging

### PR DESCRIPTION
**Pull request by imegao**

### CODE REVIEW

File: integrate/yig.sql [hunk 1]
```diff
@@ -26,6 +26,7 @@ CREATE TABLE `buckets` (
   `bucketname` varchar(255) NOT NULL DEFAULT '',
   `acl` JSON DEFAULT NULL,
   `cors` JSON DEFAULT NULL,
+  `bl` JSON NOT NULL DEFAULT '',
```
Comment (cxt90730) [inline integrate/yig.sql]: logging

File: meta/client/tidbclient/bucket.go [hunk 2]
```diff
@@ -14,13 +14,14 @@ import (
 )
 
 func (t *TidbClient) GetBucket(bucketName string) (bucket *Bucket, err error) {
-	var acl, cors, lc, policy, website, encryption, createTime string
-	sqltext := "select bucketname,acl,cors,lc,uid,policy,website,COALESCE(encryption,\"\"),createtime,usages,versioning from buckets where bucketname=?;"
+	var acl, cors, bl, lc, policy, website, encryption, createTime string
```
Comment (cxt90730) [inline meta/client/tidbclient/bucket.go]: logging not bl

File: error/api-errors.go [hunk 3]
```diff
@@ -150,6 +150,7 @@ const (
 	ErrInvalidAcl
 	ErrUnsupportedAcl
 	ErrNonUTF8Encode
+	ErrInvalidBl
```
Comment (cxt90730) [inline error/api-errors.go]: InvalidBucketLogging

File: storage/bucket.go [hunk 4]
```diff
@@ -93,6 +93,43 @@ func (yig *YigStorage) SetBucketAcl(bucketName string, policy datatype.AccessCon
...
+func (yig *YigStorage) SetBucketLogging(bucketName string, bl datatype.BucketLoggingStatus,
+	credential common.Credential) error {
+	helper.Logger.Info("enter SetBucketLogging")
+	bucket, err := yig.MetaStorage.GetBucket(bucketName, true)
+	if err != nil {
+		return err
+	}
+	if bucket.OwnerId != credential.UserId {
+		return ErrBucketAccessForbidden
+	}
```
Comment (cannium) [inline storage/bucket.go]: I believe bucket info could be fetched in API layer by means of https://github.com/journeymidnight/yig/blob/9a74b969e98983047e7303dcf28b0104e289cf34/api/object-handlers.go#L91-L95
So check bucket owner there. Same for `GetBucketLogging`.

**Pull request merged**